plugins/gtav/Game.cpp

```diff
@@ -0,0 +1,170 @@
+	const auto iter        = modules.find("GTA5.exe");
+	if (iter == modules.cend()) {
+		return MUMBLE_PDEC_ERROR_TEMP;
+	}
+	// Find out the offset of the memory region
+	const MemoryRegions &regions = iter->second.regions();
+	const auto &r                = regions.cbegin();
+	if (r != regions.cend()) {
+		m_moduleBase = r->address;
+	}
+
+	// Check if we can get meaningful data from it
+	float apos[3], afront[3], atop[3], cpos[3], cfront[3], ctop[3];
+	const char *identity = "";
+	const char *context  = "";
+
+	return fetchPositionalData(apos, afront, atop, cpos, cfront, ctop, &context, &identity);
+}
+
+Mumble_PositionalDataErrorCode Game::fetchPositionalData(float *avatarPos, float *avatarDir, float *avatarAxis,
+														 float *cameraPos, float *cameraDir, float *cameraAxis,
+														 const char **contextPtr, const char **identityPtr) {
+	for (uint8_t i = 0; i < 3; ++i) {
+		avatarPos[i] = avatarDir[i] = avatarAxis[i] = cameraPos[i] = cameraDir[i] = cameraAxis[i] = 0.0f;
+	}
+
+	// Memory addresses
+	procptr_t state_address, avatarPos_address, cameraPos_address, avatar_base_address, avatar_base, avatarDir_address,
+		avatarAxis_address, cameraDir_address, cameraAxis_address, player_address, vehicle_address, location_address,
+		street_address;
+	const std::string expected_name = "Grand Theft Auto V";
+
+	if (!m_proc.isOk()) {
+		return MUMBLE_PDEC_ERROR_TEMP;
+	}
+
+	if (m_proc.peekString(m_moduleBase + 0x1901130, expected_name.length()) == expected_name) {
+		// Steam version
+		state_address      = m_moduleBase + 0x291B2F0;
+		avatarPos_address  = m_moduleBase + 0x1D1D850;
+		cameraPos_address  = m_moduleBase + 0x1DACEB0;
+		avatarDir_address  = m_moduleBase + 0x20476D0;
+		avatarAxis_address = m_moduleBase + 0x20476A0;
+		cameraDir_address  = m_moduleBase + 0x1FA7A80;
+		cameraAxis_address = m_moduleBase + 0x1FA7A90;
+		player_address     = m_moduleBase + 0x2928674;
+		vehicle_address    = m_moduleBase + 0x2503F10;
+		location_address   = m_moduleBase + 0x25038CB;
+		street_address     = m_moduleBase + 0x2500600;
+	} else if (m_proc.peekString(0x180D4D8, expected_name.length()) == expected_name) {
+		// Retail version
+		state_address       = m_moduleBase + 0x2733490;
+		avatarPos_address   = m_moduleBase + 0x1F7EAA0;
+		cameraPos_address   = m_moduleBase + 0x1C58630;
+		avatar_base_address = m_moduleBase + 0x1B956C0;
+		// Avatar pointer
+		avatar_base = m_proc.peekPtr(avatar_base_address);
+		if (!avatar_base)
+			return MUMBLE_PDEC_ERROR_TEMP;
+		avatarDir_address  = avatar_base + 0x70;
+		avatarAxis_address = avatar_base + 0x80;
+		cameraDir_address  = m_moduleBase + 0x1C5A0F0;
+		cameraAxis_address = m_moduleBase + 0x1F7D9F0;
+		player_address     = m_moduleBase + 0x273DBAC;
+		vehicle_address    = m_moduleBase + 0x2331890;
+		location_address   = m_moduleBase + 0x233125B;
+		street_address     = m_moduleBase + 0x232DFA0;
+	} else {
+		// Return permanent error on unknown version
+		return MUMBLE_PDEC_ERROR_PERM;
+	}
```

Comment:
I would move this block in `Game::init()` (and move the variables into the class itself).